### ISSUE TITLE
Annotations in typing.Annotated are not checked

### ISSUE THREAD

**Issue by NiklasRosenstein** [python/mypy#10872]
**Bug Report**

**To Reproduce**

```py
import typing_extensions as te
x: te.Annotated[int, Foobar] = 42
```

**Expected Behavior**

Mypy complains about `Foobar` not being defined. In other cases, e.g. when a function is called in an Annotated argument, it checks if the arguments passed to the function are valid (or number of arguments match, etc. etc.).

**Actual Behavior**

Mypy does not complain about `Foobar` not being defined.

**Your Environment**

- Mypy version used: 0.910
- Mypy command-line flags: n/a
- Mypy configuration options from `mypy.ini` (and other config files): n/a
- Python version used: 3.8.2
- Operating system and version: OSX

**Comment by hauntsaninja** [comment]
I think this is a result of https://github.com/python/mypy/pull/9625 
People even want this non-checking to be extended to all usages of Annotated, e.g., see https://github.com/python/mypy/pull/10777

Fwiw, my reading of PEP 593 is that mypy should complain about your example, but not if `from __future__ import annotations`, since then it won't fail at runtime and falls more under the category of "Unknown annotations" which PEP 593 says type checkers should ignore.

**Comment by Dreamsorcerer** [comment]
This will become particularly useful with Pydantic's pipeline API:
```
v = validate_as(int).transform(str.lower)  # Type error here

class MyDict(TypedDict):
    a: Annotated[str, validate_as(int).transform(str.lower)]  # No type error here
```
